src/openms/source/CHEMISTRY/SimpleTSGXLMS.cpp

```diff
@@ -631,7 +589,13 @@ namespace OpenMS
       addPrecursorPeaks_(spectrum, precursor_mass, maxcharge);
     }
 
-    std::sort(spectrum.begin(), spectrum.end(), SimpleTSGXLMS::SimplePeakComparator());
+#ifdef OPENMS_USE_PDQSORT
+    std::reverse(spectrum.begin(), spectrum.end());
```

Comment:
same here - reverse iterator

Reply:
since the comparator is branchless, calling `boost::sort::pdqsort_branchless` should help as well.
https://www.boost.org/doc/libs/1_67_0/libs/sort/doc/html/sort/single_thread/pdqsort/pdqsort_usage.html

Reply:
using a benchmark run that takes about 35s, I don't see a difference using the branchless version. If there is one, it is smaller than the difference between repeat trials. It reached a lower minimal runtime by about 0.3s after 5 trials, but on average the difference is very small.
But the doc says it will not be slower than normal pdqsort in my case, so I will keep it in.